src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -58,8 +58,18 @@ public void onBlockPlaceExisting(BlockPlaceEvent e) {
         // This prevents Players from placing a block where another block already exists
         // While this can cause ghost blocks it also prevents them from replacing grass
         // or saplings etc...
-        if (BlockStorage.hasBlockInfo(e.getBlock())) {
-            e.setCancelled(true);
+        Block block = e.getBlock();
+        SlimefunItem sfItem = BlockStorage.check(block);
+
+        if (sfItem != null) {
+            if (e.getBlockReplacedState().getType().isAir()) {
+                for (ItemStack item : sfItem.getDrops()) {
+                    if (item != null && !item.getType().isAir()) { block.getWorld().dropItemNaturally(block.getLocation(), item); }
```

Comment:
Formatting, also for non-blocks a simple AIR check is sufficient.
```suggestion
                    if (item != null && item.getType() != Material.AIR) {
                        block.getWorld().dropItemNaturally(block.getLocation(), item);
                    }
```

Reply:
Well.... There's multiple different airs now (1.16 api and 1.17 implementation).

isAir is definitely best to use unless we know 100% that it would be AIR and not something like CAVE_AIR